packages/api/src/controllers/asset.ts

```diff
@@ -105,6 +115,40 @@ function withPlaybackUrls(asset: WithID<Asset>, ingest: string): WithID<Asset> {
   };
 }
 
+async function reconcileAssetStorage(
+  { taskScheduler }: Request,
+  asset: WithID<Asset>,
+  newStorage: Asset["storage"],
+  task?: WithID<Task>
+): Promise<{ storage: Asset["storage"]; status: Asset["status"] }> {
+  let { storage, status } = asset;
+  const ipfsParamsEq =
+    JSON.stringify(newStorage.ipfs) === JSON.stringify(storage?.ipfs);
+  if (!ipfsParamsEq) {
+    if (!newStorage.ipfs) {
+      throw new BadRequestError("Cannot remove asset from IPFS");
+    }
+    if (!task) {
+      task = await taskScheduler.scheduleTask(
```

Comment:
So, here it spawns a new export task with the new IPFS metadata. I suppose the only thing you could change is the metadata CID as the video file CID would be always the same for the same asset, does this mean we are redoing the full export on the same CID reuploading the file?